src/decomp.jl

```diff
@@ -394,14 +408,27 @@ function LinearAlgebra.factorize(A::ITensor,
   # Determines when to use eigen vs. svd (eigen is less precise,
   # so eigen should only be used if a larger cutoff is requested)
   automatic_cutoff = 1e-12
-  if which_decomp == "svd" || 
-     (isnothing(which_decomp) && cutoff ≤ automatic_cutoff)
+
+  if isnothing(which_decomp)
+    if cutoff==0.0 && !hasqns(A) && ortho != "none"
```

Comment:
We should also add a check for `maxdim` here. Possibly we could check if `maxdim <= min(dim(Lis), dim(Ris))` to see if `maxdim` would lead to a truncation.

A subtle point is that with `cutoff == 0.0`, the spectrum might have zero values, which we would want to truncate. A better code pattern would be to have the `cutoff` default to `nothing`, and then check if it has been set using `isnothing(cutoff)`, and if it has then we don't call `factorize_qr`.

We could turn this into a function like `might_truncate(m::Int, n::Int; cutoff, maxdim, ...) -> Bool` which takes keyword arguments that are used for possibly truncating and return `true` if a truncation might occur (`m` and `n` would be the matrix dimensions). Whether or not a truncation might happen is a common concept that can arise throughout the library.

In C++ ITensor we also have a keyword argument, I think called `truncate`, which you can explicitly set to `true` or `false`. If it is false, it ignores all of the keyword arguments related to truncation and does not truncate, while if it is true then it would use `cutoff = 0.0` if `cutoff` wasn't set. The logic starts to get kind of complicated, I can do another PR adding a `might_truncate(...)` function and a `truncate` keyword argument if you can make the fixes for `maxdim` and `cutoff == 0.0` here.

Reply:
I added a check for `maxdim` and changed the default value of `cutoff` to `nothing` in the latest commit.